### PR TITLE
[CAS-1007] Docs - Action to add dokka to docusaurus

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
[CAS-1007](https://stream-io.atlassian.net/browse/CAS-1007)

### 🎯 Goal

The goal is to create a pull request with the new dokka documentation in docusaurus every time we publish a new version of our SDK. 

### 🛠 Implementation details 

Added action and `scripts/dokkasaurus.sh` that manages the process to create the dokka documentation inside the docusaurus folder. 

### 🎨 UI Changes

Screenshot 2021-05-12 at 17 21 45

### 🧪 Testing

Run this action. 

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- ~[ ] Changelog is updated with client-facing changes~
- ~[ ] New code is covered by unit tests~
- ~[ ] Comparison screenshots added for visual changes~
- ~[ ] Affected documentation updated (CMS, cookbooks, tutorial)~
- [x] Reviewers added

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

File: scripts/dokkasaurus.sh
Comment (JcMinarro) [inline scripts/dokkasaurus.sh:2]: What about to use a concrete version of this repository?
We can create some tags on this GH Repository and be sure we are using this version

File: .github/workflows/publish-api-dokkasaurus.yml
Comment (JcMinarro) [inline .github/workflows/publish-api-dokkasaurus.yml:6]: We should run this action on every PR (Before we merge code into develop), and then add channes as a new commit to the PR

**Review by JcMinarro — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-api-dokkasaurus.yml
Comment (leandroBorgesFerreira) [reply .github/workflows/publish-api-dokkasaurus.yml:6]: This is a bit dangerous... It may reach a state that it enters a loop. The CI commits in the branch, then it starts again... Do some change that changes the code and restarts again... and again... and again... Although it should probably stop once there's no changes in the documentation, there's always the risk to enter a loop. Perhaps we should do it on every merge in `develop`?

**Comment by leandroBorgesFerreira** [comment]
Before merging this, we need to deploy Dokkasaurus to a package manager (like mavenCentral, JCenter...)

File: .github/workflows/publish-api-dokkasaurus.yml
Comment (JcMinarro) [reply .github/workflows/publish-api-dokkasaurus.yml:6]: What about to have a check here that see if there is any needed update on the docs?
We can block the PR is the docs is not up-to-date, asking us (devs) to update it before merge the PR (Mainly the same we do with `./gradlew apiDump`).

The reason to do that is that we need to be sure code go to develop is ready to be released anytime. We can't merge code to develop that needs some changes before it is public released into a new version.

We should have some script (Maybe the one that you did on this PR) to update the docs locally from our machine before we push it to GH

Comment (leandroBorgesFerreira) [reply .github/workflows/publish-api-dokkasaurus.yml]: Now there's only a check that the documentation is up-to-date. The developers now can ran the ./script/dokkasaurus.sh

File: scripts/dokkasaurus-ci-check.sh
Comment (JcMinarro) [inline scripts/dokkasaurus-ci-check.sh:22]: Could we add a message about what is needed to be done to fix it?

**Review by JcMinarro — CHANGES_REQUESTED** [review]

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**
